File: 0771-jewels-and-stones/0771-jewels-and-stones.java
class Solution {
    public int numJewelsInStones(String jewels, String stones) {
        Map<Character, Integer> freqs = new HashMap<>();
        
        for(char s : stones.toCharArray()) {
            // 만약 이미 계산한 돌이면 기존값 + 1
            if(freqs.containsKey(s)) {
                freqs.put(s, freqs.get(s) + 1);
            }

            // 처음 보는 돌이면 hash에 추가
            else {
                freqs.put(s, 1);
            }
        }
        
        int count = 0;
        for(char j : jewels.toCharArray()) {
            // 보석과 일치하는 돌의 개수를 합산
            if(freqs.containsKey(j)) {
                count += freqs.get(j);
            }
        }

        return count;
    }
}

File: 0771-jewels-and-stones/0771-jewels-and-stones.kt
class Solution {
    fun numJewelsInStones(jewels: String, stones: String): Int {
        var freqs: HashSet<Char> = HashSet()
        
        // 보석 jewels 종류 저장
        for(j in jewels.toCharArray()) {
            freqs.add(j)
        }
        
        // stones이 jewels인 경우
        var count = 0
        for(s in stones.toCharArray()) {
            if(freqs.contains(s)) {
                count++
            }
        }
        
        return count
    }
}